### ISSUE TITLE
Make containers named and instantiable

### ISSUE THREAD

**Issue by slavafomin** [typestack/typedi#13]
As discussed here: https://github.com/pleerock/typedi/pull/11#discussion_r110646471 we want to move from a single static container to a named instantiable ones.

Here's the desired API:

```ts
// Returns container instance of the default container:
const container1 = Container.get();

// Returns container instance of the second container.
// The additional container which goes under the "second-container" name.
const container2 = Container.get("second-container");

// This service belongs to the default container.
@Service()
class Foo {
}

// Or more meaningful API:
@Service({
  name: "bar",
  container: "second-container"
})
class Bar {
}
```

**Comment by pleerock** [comment]
Api looks good. Btw is there a reason to call `Container.create`? I think no, because its extra operation which do not do anything - named container can be created when you put `@Service` with custom container name.

**Comment by slavafomin** [comment]
@pleerock Yes, I was thinking the same. I've replaced `create()` with `get()` above. Is it ok?

**Comment by pleerock** [comment]
We were talking about separate containers to prevent "clear" method creation in tests, but we anyway endup doing that :rofl: 

All what we did - we created "named containers". I believe this feature will be rarely used.  

I think instead of `Container.get("container-name").get(Service)` we just do `Container.get(service: string|Function, containerName: string = "default")`

**Comment by pleerock** [comment]
do you have in plans to work on this issue? - so we can release everything together in next 0.5.0 version.

**Comment by slavafomin** [comment]
@pleerock I think we should leave it for now. I'm personally not requiring this functionality at the moment. Lets see if someone needs it. @all please feel free to vote on the first post in this thread.

**Comment by pleerock** [comment]
This feature request was implemented and will be released in `0.6.0`. Example of usage:

```ts
const controller1 = Container.of("mycontainer1").get(QuestionController);
controller1.save("Timber");
Container.reset("mycontainer1");

const controller2 = Container.of("mycontainer2").get(QuestionController);
controller2.save("Slava"); // different controller instance
Container.reset("mycontainer2");
```
